### PR TITLE
Improve tool usage in the cook and other crafters

### PR DESCRIPTION
**Pull request by Mekle001**
# Changes proposed in this pull request:
- fix tool usage for larger stacks during crafting
- add haybale decompression to farmer
- bit of cleanup here and there. 
- Make compost a bit cheaper
- fix ingredient resolution for cookable foods - prestock of cookables should work now. 

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/util/FurnaceRecipes.java [hunk 1]
```diff
@@ -55,16 +54,13 @@ private void loadRecipes(final MinecraftServer server)
...
 
-                    recipes.put(new ItemStorage(stack), storage);
+                recipes.put(storage.getCleanedInput().get(0), storage);
 
-                    final ItemStack output = recipe.getRecipeOutput();
-                    output.setCount(1);
-                    reverseRecipes.put(new ItemStorage(output), storage);
-                }
+                final ItemStack output = recipe.getRecipeOutput();
+                output.setCount(1);
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/util/FurnaceRecipes.java:62]: I think you want to copy this stack before modifying it? else the "recipe" objects output stack might get changed for further uses

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingCook.java [hunk 2]
```diff
@@ -103,6 +104,7 @@ public BuildingCook(final IColony c, final BlockPos l)
         keepX.put(ItemStackUtils.ISFOOD, new Tuple<>(STACKSIZE, true));
         keepX.put(ItemStackUtils.ISCOOKABLE, new Tuple<>(STACKSIZE, true));
         keepX.put(stack -> isAllowedFuel(stack), new Tuple<>(STACKSIZE, true));
+        keepX.put(stack -> !ItemStackUtils.isEmpty(stack.getContainerItem()), new Tuple<>(STACKSIZE, false));
     }
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingCook.java:109]: Will the cook keep all container items for forever? 20 buckets of milk or water too?

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java [hunk 3]
```diff
@@ -271,16 +271,16 @@ protected IAIState checkForImportantJobs()
             citizenToServe.addAll(citizenList);
             playerToServe.addAll(playerList);
 
-            if (InventoryUtils.hasItemInItemHandler(worker.getInventoryCitizen(), ItemStackUtils.CAN_EAT))
+            if (InventoryUtils.hasItemInItemHandler(worker.getInventoryCitizen(), i -> ItemStackUtils.CAN_EAT.test(i) && !getOwnBuilding().isItemStackInRequest(i)))
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java]: "i" might be bad, better stack

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingCook.java (hunk 2 above)
Comment (Mekle001) [reply src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingCook.java:109]: I should probably exclude milk and water. Probably by excluding buckets. I want to replace this with the custom data load you and I talked about eventually, but realized that this was pretty close to what was needed for now.

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/ldtteam/minecolonies/pull-request/5728/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ldtteam/minecolonies)

**Review by someaddons — APPROVED** [review]

**Pull request merged**